controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -250,7 +351,8 @@ private boolean greaterThan(Map<Long, Long> cut1, Map<StreamSegmentRecord, Integ
      */
     @Override
     public CompletableFuture<Void> startUpdateConfiguration(final StreamConfiguration newConfiguration) {
-        return getVersionedConfigurationRecord()
+        return createSubscribersRecordIfAbsent()
```

Comment:
With this the SubscribersSet record gets added only when updateStream() is invoked.
However,  if the client tries to addSubscriber() to a Stream, without changing the retention policy using updateStream, it will lead to an exception.
Perhaps it may be good to add `createSubscribersRecordIfAbsent()` here:
https://github.com/pravega/pravega/blob/master/controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java#L528

Reply:
thats a good suggestion. it keeps the change in context. i will also add the comment that this is only for streams prior to 0.9 being changed to CBR policy